pkg/storage/storage.go

```diff
@@ -159,24 +167,12 @@ func (s *storageModule) initialize(policy pkg.StoragePolicy) error {
 		return err
 	}
 
-	for _, volume := range existingPools {
-		if _, mounted := volume.Mounted(); mounted {
-			log.Debug().Msgf("Volume %s already mounted", volume.Name())
-			// volume is already mounted, skip mounting it again, make sure it is
-			// in the list of available pools. Since we are in the initialize method,
-			// we can safely assume the pool has not been added before, so no need
-			// to check for duplicate entries.
-			s.volumes = append(s.volumes, volume)
-			continue
-		}
-		_, err = volume.Mount()
+	for _, pool := range existingPools {
+		_, err := pool.Mount()
 		if err != nil {
-			s.brokenPools = append(s.brokenPools, pkg.BrokenPool{Label: volume.Name(), Err: err})
-			log.Warn().Msgf("Failed to mount volume %v", volume.Name())
-			continue
+			return err
```

Comment:
Why don't we keep track of the broken pools anymore ? I don't think you can get rid of this.